File: 0501_0600/0530/0530.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    w, h = (int(x) for x in f.readline().split())
    s = [f.readline().strip() for i in range(h)]
    t = [f.readline().strip() for i in range(h)]
    p = f.readline().strip()
    q.write('\n'.join(''.join(p[0] if s[i][j] == '0' and t[i][j] == '0' else p[1] if s[i][j] == '0' and t[i][j] == '1' else p[2] if s[i][j] == '1' and t[i][j] == '0' else p[3] for j in range(w)) for i in range(h)))
        

File: 0501_0600/0530/0530.c
#include <stdio.h>
#include <ctype.h>
#include <stdlib.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int w, h;
	char *s, *t, c, p[5];

	fscanf(f, "%d%d", &w, &h);
	s = (char*)malloc(w * h);
	t = (char*)malloc(w * h);
	for (int i = 0; i < w * h; ++i)
	{
		while (isspace(c = getc(f)));
		s[i] = c - '0';
	}
	for (int i = 0; i < w * h; ++i)
	{
		while (isspace(c = getc(f)));
		t[i] = c - '0';
	}
	fscanf(f, "%s", p);
	for (int i = 0; i < w * h; ++i)
		s[i] = (!s[i] && !t[i] ? p[0] : !s[i] && t[i] ? p[1] : s[i] && !t[i] ? p[2] : p[3]);
	for (int i = 0; i < h * w; i += w)
		fprintf(q, "%.*s\n", w, s + i);
	return 0;
}
